MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -4726,12 +4729,14 @@ static uint32_t mvkGetEntryProperty(io_registry_entry_t entry, CFStringRef prope
 				} else {
 					pImgInfo->mtlTexture = mvkBuffView->getMTLTexture();
 				}
+				[pImgInfo->mtlTexture retain];
 				break;
 			}
 			case VK_STRUCTURE_TYPE_EXPORT_METAL_IO_SURFACE_INFO_EXT: {
 				auto* pIOSurfInfo = (VkExportMetalIOSurfaceInfoEXT*)next;
 				auto* mvkImg = (MVKImage*)pIOSurfInfo->image;
 				pIOSurfInfo->ioSurface = mvkImg->getIOSurface();
+				CFRetain(pIOSurfInfo->ioSurface);
```

Comment:
This is a curious case here. `IOSurface` was originally a Core Foundation-style object, which means that you had to use manual retain/release even from ARC-enabled Objective-C. But now there's an actual `IOSurface` class. Further, Swift is capable of automatically managing Core Foundation objects, and I believe it will actually wind up importing any `IOSurfaceRef` as an instance of the Objective-C `IOSurface` class. So, you really do need the `CFRetain()` here for Swift. (From Objective-C that means that the caller would either have to `CFRelease()` it manually, or with ARC use a `__bridge_transfer` cast or a call to `CFBridgingRelease()` to tell ARC about it.)